src/server/utilities/signed_url.rs

```diff
@@ -38,146 +32,200 @@ impl FromStr for Platform {
     fn from_str(input: &str) -> std::result::Result<Self, Self::Err> {
         let url = Url::parse(input).context("failed to parse URL")?;
         match url.scheme() {
-            "s3" => Ok(Self::Aws {
-                url: String::from(url.as_str()),
-                bucket: String::from(url.domain().unwrap_or("")),
-                path: String::from(url.path().strip_prefix('/').unwrap_or("")),
-            }),
-            "s3a" => Ok(Self::Aws {
-                url: String::from(url.as_str()),
-                bucket: String::from(url.domain().unwrap_or("")),
-                path: String::from(url.path().strip_prefix('/').unwrap_or("")),
-            }),
-            "gs" => Ok(Self::Gcp {
-                url: String::from(url.as_str()),
-                bucket: String::from(url.domain().unwrap_or("")),
-                path: String::from(url.path().strip_prefix('/').unwrap_or("")),
-            }),
-            _ => Ok(Self::None {
-                url: String::from(url.as_str()),
-            }),
+            "s3" | "s3a" => Ok(Self::Aws),
+            "abfss" | "abfs" => Ok(Self::Azure),
```

Comment:
had interesting discussions around this in the past :D - essentially in azure there are many ways a scheme could look like, some "official" some de-facto standards (e.g. what `fsspec` adopted).. Also onelake did not make things less confusing.

Technically `abfs[s]` refers to a specific implementation of an hdfs compliant driver...

In the end its fine to do this for now, I'll eave some comments later around the general signing question.